src/Microsoft.Extensions.Logging.Console/Internal/ConsoleLoggerProcessor.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Logging.Console.Internal
+{
+    public class ConsoleLoggerProcessor
+    {
+        private const int _maxQueuedMessages = 1024;
+        // Writing to console is not an atomic operation in the current implementation and since multiple logger 
+        // instances are created with a different name. Also since Console is global, using a static lock is fine. 
+        private static readonly object _lock = new object();
+
+        private IConsole _console;
+
+        private readonly BlockingCollection<LogMessageEntry> _messageQueue = new BlockingCollection<LogMessageEntry>(_maxQueuedMessages);
+        private readonly Task _outputTask;
+
+        public ConsoleLoggerProcessor()
+        {
+            RegisterForExit();
+
+            // Start Console message queue processor
+            _outputTask = Task.Factory.StartNew(
+                ProcessLogQueue,
+                this,
+                TaskCreationOptions.LongRunning);
+        }
+
+        public IConsole Console
+        {
+            get { return _console; }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                _console = value;
+            }
+        }
+
+        public bool HasQueuedMessages => _messageQueue.Count > 0;
```

Comment:
`BlockingCollection` doesn't expose `.IsEmpty` from the underlying `ConcurrentQueue`; but testing against `.Count > 0` is a lot more expensive.

Wonder if you should use the explicit ` BlockingCollection(IProducerConsumerCollection<T> collection, int boundedCapacity)` ctor and pass in as the [ctor you are calling does](https://github.com/dotnet/corefx/blob/master/src/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L160-L161); but keep a reference to a ConcurrentQueue for the fast `IsEmpty` test?

Or not expose the new `HasQueuedMessages` property on ConsoleLogger...

/cc @stephentoub